### ISSUE TITLE
Add benchmarks to measure performance of `evaluate_function_over_domain` function

### ISSUE THREAD

**Issue by z2oh** [z2oh/sexe#31]
Documentation for adding benchmarking tests can be found here: https://doc.rust-lang.org/1.8.0/book/benchmark-tests.html

We should pick a few functions to benchmark. Some good candidates are

* `x`
* `sin(x)`
* `|sin(x^x) / 2^((x^x - pi/2)/pi)|`

Feel free to add more functions to this list. We should also try a variety of resolutions (as low as `100` and as high as `1_000_000_000`).

These benchmarks should be added to the interface module, since that is where the `evaluate_function_over_domain` function resides.

**Comment by Istar-Eldritch** [comment]
Hey @z2oh I'll give this a try.

**Comment by z2oh** [comment]
Hi @Istar-Eldritch, thanks for your interest! There is already a PR which implements this as described here: see conversation here #33 . However, due to some limitations with current benchmarking in Rust, some project rearchitecting has to take place in order for that to happen. If you want to give that a try, it would be appreciated!

**Comment by Istar-Eldritch** [comment]
@z2oh It may be worth to create a master issue to keep track of the refactoring. We can continue the discussion from where you left in #33 there. We should do it in multiple stages to simplify the review process.

**Comment by z2oh** [comment]
@Istar-Eldritch I agree. I made #35 just now to track the refactoring, then we can come back to this issue to track benchmarking.
